packages/query/src/index.ts

```diff
@@ -1126,10 +1129,12 @@ const generateQueryHook = async (
       'implementation',
     );
 
-    const routeString = `\`${route}\``;
+    const routeString = isVue(outputClient)
+      ? getRouteAsArray(route) // Note: this is required for reactivity to work, we will lose it if route params are converted into string, only as array they will be tracked // TODO: add tests for this
+      : `\`${route}\``;
 
+    // Note: do not unref() params in Vue - this will make key lose reactivity
     const queryKeyFn = `export const ${queryKeyFnName} = (${queryKeyProps}) => {
-    ${isVue(outputClient) ? vueUnRefParams(props) : ''}
```

Comment:
@Maxim-Mazurok, thanks for the detailed description of the issue here. Sorry for the regression added to main. 
The `unref` in query key was a consequence of accidentally reverting to a string based route key. 

Preserving the refs does pose a challenge for the useNamedParameters feature. I'm onboard with your proposal to this feature for vue since there was never explicitly a request for it. We added it just to have uniform support across all clients. Do you think it's generally not useful in the case of vue or just a challenge to implement right now?

Do you think `MaybeRefDeep` would work in the case of named parameters? 

```ts
export const getListPetsQueryKey = (
  pathParams: MaybeRefDeep<ListPetsPathParameters> = {},
  params?: MaybeRef<ListPetsParams>,
) => {
  const { version } = cloneDeepUnref(pathParams);
  return ['v', version, 'pets', ...(params ? [params] : [])] as const;
};
```

Reply:
Got it, all good :) We should've added tests in the first place :)

Thanks for being on board with disabling named params for Vue for now. I think it might be equally useful for Vue as for any other framework, it's just a technical challenge I believe.

`MaybeRefDeep` seems to work type-wise, however, it didn't pass the test I've added to verify path params reactivity. You can check it out in this commit: [`a36a767` (#931)](https://github.com/anymaniax/orval/pull/931/commits/a36a767e4300399d4cad65081ffbc7f14b9d8c9f)

> Would it be more idiomatic to have a `computed` statement for each path parameter or one for all the parameters?

I'm not sure... I'd expect both to work. It's pretty annoying to deal with all that reactivity stuff, you literally have to check all the options...

---

I will disable named params for vue-query client for now in this PR